File: trail2/재귀함수를 이용한 최소공배수/least-common-multiple-using-recursive-function.cpp
#include <iostream>

using namespace std;

int n;
int arr[10];

int GetGCD(int a, int b) {
    if (b == 0)
        return a;
    
    return GetGCD(b, a % b);
}

int GetLCMOfTwo(int a, int b) {
    return (a * b) / GetGCD(a, b);
}

int GetLCMOfArray(int idx) {
    if (idx == 0)
        return arr[0];

    return GetLCMOfTwo(arr[idx], GetLCMOfArray(idx - 1));
}

int main() {
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    // Please write your code here.
    cout << GetLCMOfArray(n - 1) << endl;

    return 0;
}

File: trail2/재귀함수를 이용한 최소공배수/least-common-multiple-using-recursive-function.py
n = int(input())
arr = list(map(int, input().split()))

def get_gcd(a, b):
    if b == 0:
        return a
    
    return get_gcd(b, a % b)

def get_lcm_two(a, b):
    return (a * b) // get_gcd(a, b)

def get_lcm_array(n):
    if n == 0:
        return arr[0]
    
    return get_lcm_two(arr[n], get_lcm_array(n - 1))

print(get_lcm_array(n-1))